lib/Mojo/mysql/Connection.pm

```diff
@@ -0,0 +1,797 @@
+
+  $self->emit('connect') if $self->_state eq 'auth';
+  $self->emit('end') if $self->_state eq 'query';
+  $self->_state('idle');
+}
+
+sub _recv_query_responce {
+  my $self = shift;
+  my $first = $self->_get_int(1);
+  return $self->_recv_error if $first == 255;
+  return $self->_recv_ok if $first == 0;
+
+  $self->{field_count} = $self->_chew_lcint;
+
+  warn '<<< QUERY_RESPONSE ', $self->{connection_id}, ' #', $self->{seq}, ' state:', $self->_state, "\n",
+    ' fields:', $self->{field_count}, "\n" if DEBUG;
+
+  $self->_state('field');
+}
+
+sub _recv_eof {
+  my $self = shift;
+  my $first = $self->_get_int(1);
+  return $self->_recv_error if $first == 255;
+  die "_recv_eof() wrong packet $first" unless $first == 254;
+
+  $self->_chew_int(1);
+  $self->{warnings_count} = $self->_chew_int(2);
+  $self->{status_flags} = $self->_chew_int(2);
+
+  warn '<<< EOF ', $self->{connection_id}, ' #', $self->{seq}, ' state:', $self->_state, "\n",
+    ' warnings:', $self->{warnings_count},
+    ' status:', _flag_list(SERVER_STATUS, $self->{status_flags}),
+    '(', sprintf('%04X', $self->{status_flags}), ')', "\n" if DEBUG;
+
+  if ($self->_state eq 'field') {
+    $self->emit(fields => $self->{column_info});
+    $self->_state('result');
+  }
+  elsif ($self->_state eq 'result') {
+    $self->{column_info} = [];
+    if ($self->{status_flags} & 0x0008) {
+      # MORE_RESULTS
+      $self->_state('query');
+    }
+    else {
+      $self->emit(end => undef);
+      $self->_state('idle');
+    }
+  }
+}
+
+sub _recv_field {
+  my $self = shift;
+  my $first = $self->_get_int(1);
+  return $self->_recv_error if $first == 255;
+  return $self->_recv_eof if $first == 254;
+  die "_recv_field() wrong packet $first" if $first > 250;
+
+  my $field = {};
+  $field->{catalog} = $self->_chew_lcstr;
+  $field->{schema} = $self->_chew_lcstr;
+  $field->{table} = $self->_chew_lcstr;
+  $field->{org_table} = $self->_chew_lcstr;
+  $field->{name} = $self->_chew_lcstr;
+  $field->{org_name} = $self->_chew_lcstr;
+  $self->_chew_lcint;
+  $field->{character_set} = $self->_chew_int(2);
+  $field->{column_length} = $self->_chew_int(4);
+  $field->{column_type} = $self->_chew_int(1);
+  $field->{flags} = $self->_chew_int(2);
+  $field->{decimals} = $self->_chew_int(1);
+  $self->_chew_str(2);
+
+  do { _utf8_on $field->{$_} for qw(catalog schema table org_table name org_name) } if $self->options->{utf8};
+
+  push @{$self->{column_info}}, $field;
+
+  warn '<<< FIELD ', $self->{connection_id}, ' #', $self->{seq}, ' state:', $self->_state, "\n",
+    ' name:', $field->{name},
+    ' type:', REV_DATATYPE->{chr $field->{column_type}}, '(', $field->{column_type}, ')',
+    ' length:', $field->{column_length},
+    ' charset:', REV_CHARSET->{$field->{character_set}} // 'UNKNOWN', '(', $field->{character_set}, ')',
+    ' flags:', _flag_list(FIELD_FLAG, $field->{flags}), '(', $field->{flags}, ')', , "\n" if DEBUG;
```

Comment:
Packet tracing can really overload the terminal. Changed to be Enabled if DEBUG > 1